File: jdbc-template/src/main/java/io/thomastodon/spring/data/examples/house/HouseRepository.java
package io.thomastodon.spring.data.examples.house;

import java.util.List;
import java.util.UUID;

public interface HouseRepository {

    List<House> findAll();

    void saveHouse(House house);

    void addRoomToHouse(Room room, UUID houseUuid);

    void addChairToRoom(Chair chair, UUID roomUuid);

    void addLegToChair(Leg leg, UUID chairUuid);

    void deleteAll();
}

File: jdbc-template/src/main/java/io/thomastodon/spring/data/examples/house/HouseRepository.kt
package io.thomastodon.spring.data.examples.house

import java.util.*

interface HouseRepository {

    fun findAll(): Map<UUID, House>

    fun saveHouse(house: House): UUID

    fun addRoomToHouse(houseUuid: UUID, room: Room): UUID

    fun addChairToRoom(roomUuid: UUID, chair: Chair): UUID

    fun addLegToChair(chairUuid: UUID, leg: Leg): UUID

    fun deleteAll()
}
